### PR TITLE
arch/sim: Adjust the stack size for up_use_stack too

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
See the discussion here: https://github.com/apache/incubator-nuttx/pull/5365

## Impact
up_use_stack on sim

### CODE REVIEW

File: arch/sim/src/sim/up_usestack.c [hunk 1]
```diff
@@ -113,4 +114,11 @@ int up_use_stack(FAR struct tcb_s *tcb, FAR void *stack, size_t stack_size)
 #endif
 
   return OK;
+#else
+  /* Ignore the stack argument and allocate a new stack instead
+   * since it's impossible to extend a preallocated memory in place.
+   */
+
+  return up_create_stack(tcb, stack_size, tcb->flags & TCB_FLAG_TTYPE_MASK);
```
Comment (yamt) [inline arch/sim/src/sim/up_usestack.c:122]: `up_tls_size() + stack_size`?

Comment (xiaoxiang781216) [reply arch/sim/src/sim/up_usestack.c:122]: up_create_stack already consider this case.

Comment (yamt) [reply arch/sim/src/sim/up_usestack.c:122]: it's usually the responsibility of the caller of up_create_stack to adjust for up_tls_size, isn't it?

https://github.com/apache/incubator-nuttx/blob/a1af6ece3f6c06679c5be3ac3e9e956dc5d91a82/sched/task/task_init.c#L114-L127

Comment (xiaoxiang781216) [reply arch/sim/src/sim/up_usestack.c:122]: Yes, you are right. Done.

**Comment by xiaoxiang781216** [comment]
@yamt do you have more comment?

**Review by yamt — APPROVED** [review]
honestly speaking i still feel it's better to revert the adjustment patch. but i can live with this.

**Pull request merged**